services/app-api/src/handlers/add_sha.ts

```diff
@@ -34,62 +39,77 @@ export const calculateSHA256 = async (s3URL: string): Promise<string> => {
     } catch (err) {
-        console.error(`Error calculating SHA256 for ${s3URL}: ${err}`)
-        throw err
+        console.error(`Error in calculateSHA256 for ${s3URL}: ${err}`)
+        return ''
     }
 }
 
 export const updateDocumentsSHA256 = async (
-    documents: SubmissionDocument[]
+    documents: SubmissionDocument[],
+    serviceName: string
 ): Promise<SubmissionDocument[]> => {
     try {
-        await Promise.all(
+        const updatedDocuments = await Promise.all(
             documents.map(async (document) => {
                 if (
                     !Object.prototype.hasOwnProperty.call(document, 'sha256') ||
                     !document.sha256
                 ) {
                     try {
                         const sha256 = await calculateSHA256(document.s3URL)
-                        document.sha256 = `${sha256}`
+                        const updatedDocument = {
+                            ...document,
+                            sha256: `${sha256}`,
+                        }
+                        return updatedDocument
                     } catch (error) {
-                        console.error('Error calculating SHA256:', error)
+                        console.error('Error in updateDocumentsSHA256:', error)
+                        recordException(error, serviceName, 'calculateSHA256')
+                        // Return the original document if an error occurs
+                        return document
                     }
+                } else {
+                    return document
                 }
             })
         )
-        return documents
+        return updatedDocuments
     } catch (error) {
         console.error('Error in updateDocumentsSHA256:', error)
+        recordException(error, serviceName, 'updateDocumentsSHA256')
         throw error
     }
 }
 
 export const processRevisions = async (
     store: Store,
-    revisions: HealthPlanRevisionTable[]
+    revisions: HealthPlanRevisionTable[],
+    serviceName: string
 ): Promise<void> => {
     for (const revision of revisions) {
         const pkgID = revision.pkgID
         const decodedFormDataProto = toDomain(revision.formDataProto)
         if (!(decodedFormDataProto instanceof Error)) {
             const formData = decodedFormDataProto as HealthPlanFormDataType
-            formData.documents = await updateDocumentsSHA256(formData.documents)
+            formData.documents = await updateDocumentsSHA256(
+                formData.documents,
+                serviceName
+            )
             formData.contractDocuments = await updateDocumentsSHA256(
-                formData.contractDocuments
+                formData.contractDocuments,
+                serviceName
             )
             for (const rateInfo of formData.rateInfos) {
                 rateInfo.rateDocuments = await updateDocumentsSHA256(
-                    rateInfo.rateDocuments
+                    rateInfo.rateDocuments,
```

Comment:
Just to confirm, we will want this PR to merge in before [#1763](https://github.com/Enterprise-CMCS/managed-care-review/pull/1763)?

Otherwise, we would need to add `rateInfo.supportingDocuments` in here.